examples/list.rs

```diff
@@ -62,220 +318,30 @@ impl<T> StatefulList<T> {
-                ("Item2", 1),
-                ("Item3", 3),
-                ("Item4", 1),
-                ("Item5", 4),
-                ("Item6", 1),
-                ("Item7", 3),
-                ("Item8", 1),
-                ("Item9", 6),
-                ("Item10", 1),
-                ("Item11", 3),
-                ("Item12", 1),
-                ("Item13", 2),
-                ("Item14", 1),
-                ("Item15", 1),
-                ("Item16", 4),
-                ("Item17", 1),
-                ("Item18", 5),
-                ("Item19", 4),
-                ("Item20", 1),
-                ("Item21", 2),
-                ("Item22", 1),
-                ("Item23", 3),
-                ("Item24", 1),
-            ]),
-            events: vec![
-                ("Event1", "INFO"),
-                ("Event2", "INFO"),
-                ("Event3", "CRITICAL"),
-                ("Event4", "ERROR"),
-                ("Event5", "INFO"),
-                ("Event6", "INFO"),
-                ("Event7", "WARNING"),
-                ("Event8", "INFO"),
-                ("Event9", "INFO"),
-                ("Event10", "INFO"),
-                ("Event11", "CRITICAL"),
-                ("Event12", "INFO"),
-                ("Event13", "INFO"),
-                ("Event14", "INFO"),
-                ("Event15", "INFO"),
-                ("Event16", "INFO"),
-                ("Event17", "ERROR"),
-                ("Event18", "ERROR"),
-                ("Event19", "INFO"),
-                ("Event20", "INFO"),
-                ("Event21", "WARNING"),
-                ("Event22", "INFO"),
-                ("Event23", "INFO"),
-                ("Event24", "WARNING"),
-                ("Event25", "INFO"),
-                ("Event26", "INFO"),
-            ],
-        }
-    }
-
-    /// Rotate through the event list.
-    /// This only exists to simulate some kind of "progress"
-    fn on_tick(&mut self) {
-        let event = self.events.remove(0);
-        self.events.push(event);
-    }
-}
-
-fn main() -> Result<(), Box<dyn Error>> {
-    // setup terminal
-    enable_raw_mode()?;
-    let mut stdout = io::stdout();
-    execute!(stdout, EnterAlternateScreen, EnableMouseCapture)?;
-    let backend = CrosstermBackend::new(stdout);
-    let mut terminal = Terminal::new(backend)?;
-
-    // create app and run it
-    let tick_rate = Duration::from_millis(250);
-    let app = App::new();
-    let res = run_app(&mut terminal, app, tick_rate);
+impl TodoItem<'_> {
+    fn to_list_item(&self, index: usize) -> ListItem {
+        let bg_color = match index % 2 {
+            0 => NORMAL_ROW_COLOR,
+            _ => ALT_ROW_COLOR,
+        };
+        let line = match self.status {
+            Status::Todo => Line::styled(format!(" ☐ {}", self.todo), (TEXT_COLOR, bg_color)),
+            Status::Completed => Line::styled(
+                format!(" ✓ {}", self.todo),
+                (COMPLETED_TEXT_COLOR, bg_color),
+            ),
+        };
```

Comment:
It didn't work. 

I also tried this but also didn't work.

```
let _ = inner_area.rows().enumerate().map(|(i, line)| {
            let bg_color = match i % 2 {
                0 => NORMAL_ROW_COLOR,
                _ => ALT_ROW_COLOR,
            };
            buf.set_style(line, bg_color);
        });
```

maybe I used inner_area(Rect used for list) or buf.set_style() incorrectly. Any other ideas?

Reply:
Maybe I expressed myself wrong. You are trying to style the `Line` struct _inside_ `ListItem`, this line is rendered with an offset because of the selection. But you can style `ListItem` _directly_ like below to style the entire item.

`Line` is a _part_ of `ListItem`, whereas `ListItem` represent an entire entry in the list.

Hope that was clear enough, here is the code I was thinking of originally (it works):

```patch
diff --git a/examples/list.rs b/examples/list.rs
index 2d3ad14..e3037b7 100644
--- a/examples/list.rs
+++ b/examples/list.rs
@@ -325,14 +325,14 @@ impl TodoItem<'_> {
             _ => ALT_ROW_COLOR,
         };
         let line = match self.status {
-            Status::Todo => Line::styled(format!(" ☐ {}", self.todo), (TEXT_COLOR, bg_color)),
+            Status::Todo => Line::styled(format!(" ☐ {}", self.todo), TEXT_COLOR),
             Status::Completed => Line::styled(
                 format!(" ✓ {}", self.todo),
                 (COMPLETED_TEXT_COLOR, bg_color),
             ),
         };
 
-        ListItem::new(line)
+        ListItem::new(line).bg(bg_color)
     }
 }
 ```

Reply:
Interesting... I tried this at the beginning but apparently, I made some mistakes so it didn't work.

But thanks!